pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -687,19 +687,12 @@ func (s *DiscoveryServer) generateEndpoints(
 	}
 
 	// If locality aware routing is enabled, prioritize endpoints or set their lb weight.
-	if push.Mesh.LocalityLbSetting != nil {
-		// Make a shallow copy of the cla as we are mutating the endpoints with priorities/weights relative to the calling proxy
-		clonedCLA := util.CloneClusterLoadAssignment(l)
```

Comment:
Why do you remove this

Reply:
Good catch, this as accidental. I am surprised tests pass...

Reply:
I added it back